core/src/main/java/org/elasticsearch/index/query/GeoShapeQueryBuilder.java

```diff
@@ -183,6 +183,10 @@ public String indexedShapeType() {
      * @return this
      */
     public GeoShapeQueryBuilder strategy(SpatialStrategy strategy) {
+        if (strategy != null && strategy == SpatialStrategy.TERM && relation != ShapeRelation.INTERSECTS) {
```

Comment:
does it make sense to check for null beforehand and throw exception? we don't want anybody to call this method and explicitly pass in `null` I guess.